src/miniscript/decode.rs

```diff
@@ -173,6 +173,7 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> TerminalStack<Pk, Ctx> {
         F: FnOnce(Arc<Miniscript<Pk, Ctx>>) -> Terminal<Pk, Ctx>,
     {
         let top = self.pop().unwrap();
+        Ctx::check_ms_validity(&top)?;
```

Comment:
Oops! It turns out that this can panic if we call it on malformed input, because it tries to count the max_dissatisfaction_witness_elements but we still have a panic there https://github.com/rust-bitcoin/rust-miniscript/blob/master/src/miniscript/astelem.rs#L774

This caused a fuzztest failure on another PR.